[css-masonry] Update Intrinsic Sizing Rows 004 Mix1 Expected / Ref Test Case

stwrt opened the pull request:
#### 7fb07870ad51fa511629e16e59e70fed3d075bbe
<pre>
[css-masonry] Update Intrinsic Sizing Rows 004 Mix1 Expected / Ref Test Case
<a href="https://bugs.webkit.org/show_bug.cgi?id=283068">https://bugs.webkit.org/show_bug.cgi?id=283068</a>
<a href="https://rdar.apple.com/problem/139815982">rdar://problem/139815982</a>

Reviewed by Sammy Gill.

Use Ahem font and add colors to work around font sizing issues.

* Grid 1

Row 3 needs to span at least 2ch, because item 1 in the test case has a height of 2ch.
Therefore, item &quot;3 3&quot; should set a similar height of 2ch.

* Grid 2

Row 1 needs a height of 3ch, because in the original test case &quot;2 2&quot; and &quot;3 3&quot; will be placed in this row during track sizing.

Row 3 needs to span at least 2ch, because item 5 in the test case has a height of 2ch.
Therefore, item &quot;3 3&quot; should set a similar height of 2ch.

Item 5 needs a height of 2ch, because it is set to that in the original test case.

* Grid 6

Remove extra 5 not found in original test case.

* Grid 7

Set the grid-template-columns, because the second column will be incorrectly computed without &quot;30px auto&quot;.
Row 1 need a height of 3ch.
Row 4 needs to be a max-content, which is 3ch.
Simplify item 5 to just a span of 4.

* Grid 8

Set the grid-template-columns, because the second column will be incorrectly computed without &quot;30px auto&quot;.
Row 1 need a height of 3ch.
Row 4 needs to be a max-content, which is 3ch.
Simplify item 5 to just a span of 4 and height of 6ch.

* Grid 9

Set the grid-template-columns, because the second column will be incorrectly computed without &quot;30px auto&quot;.
Row 1 need a height of 3ch.
Row 4 needs to be a max-content, which is 3ch.
Simplify item 5 to just a span of 3 and height of 6ch.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-004-mix1-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-004-mix1-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-004-mix1.html:

Canonical link: <a href="https://commits.webkit.org/286779@main">https://commits.webkit.org/286779@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/99f3b14a89aa4537bcf7c8e25a0527141e93ca2f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/77043 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/56078 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29958 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/81593 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/28322 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/65226 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/4374 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/60368 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18440 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/80110 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/50322 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/66133 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40679 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/47724 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/23631 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/26648 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/68840 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23959 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/83027 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/4423 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/2968 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/68645 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/4578 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/66106 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67897 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16941 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11881 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/9970 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/4369 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/4389 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/7824 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/6148 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-004-mix1-expected.html:
```diff
@@ -33,27 +34,51 @@
 
 <section>
-<grid>
+<grid style="grid-template-columns: 15px auto;">
```
Can you double check if this specific change and the ones like this in the grids below are similar to the ones I commented on in https://github.com/WebKit/WebKit/pull/36546#discussion_r1841623428?

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-004-mix1-expected.html:
```diff
@@ -97,44 +122,34 @@
+    <item>3 3</item>
+    <item style="height: 3ch">4</item>
+    <item style="height: 6ch; grid-row:span 3">5 5</item>
+  </grid>
```
Spacing is off here

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-004-mix1-ref.html:
```diff
@@ -97,44 +122,34 @@
+    <item>3 3</item>
+    <item style="height: 3ch">4</item>
+    <item style="height: 6ch; grid-row:span 3">5 5</item>
+  </grid>
```
Spacing is off here

sammygill reviewed (commented):
I think the commit message needs to be updated to reflect what the changes are. For example, I don't see where in grid 1 where we are specifying column sizes. Also how are the hidden items in 7-9 adding an extra column? Aren't there visible items that are placed in the same columns they span?

sammygill reviewed (approved):
Since we're setting the column sizes in this test we should also set the font size in the style

stwrt committed:
[css-masonry] Update Intrinsic Sizing Rows 004 Mix1 Expected / Ref Test Case
https://bugs.webkit.org/show_bug.cgi?id=283068
rdar://problem/139815982

Reviewed by Sammy Gill.

Use Ahem font and add colors to work around font sizing issues.

* Grid 1

Row 3 needs to span at least 2ch, because item 1 in the test case has a height of 2ch.
Therefore, item "3 3" should set a similar height of 2ch.

* Grid 2

Row 1 needs a height of 3ch, because in the original test case "2 2" and "3 3" will be placed in this row during track sizing.

Row 3 needs to span at least 2ch, because item 5 in the test case has a height of 2ch.
Therefore, item "3 3" should set a similar height of 2ch.

Item 5 needs a height of 2ch, because it is set to that in the original test case.

* Grid 6

Remove extra 5 not found in original test case.

* Grid 7

Set the grid-template-columns, because the second column will be incorrectly computed without "30px auto".
Row 1 need a height of 3ch.
Row 4 needs to be a max-content, which is 3ch.
Simplify item 5 to just a span of 4.

* Grid 8

Set the grid-template-columns, because the second column will be incorrectly computed without "30px auto".
Row 1 need a height of 3ch.
Row 4 needs to be a max-content, which is 3ch.
Simplify item 5 to just a span of 4 and height of 6ch.

* Grid 9

Set the grid-template-columns, because the second column will be incorrectly computed without "30px auto".
Row 1 need a height of 3ch.
Row 4 needs to be a max-content, which is 3ch.
Simplify item 5 to just a span of 3 and height of 6ch.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-004-mix1-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-004-mix1-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-004-mix1.html:

Canonical link: https://commits.webkit.org/286779@main
